src/types.ts

```diff
@@ -35,16 +35,40 @@ export interface ZAuctionInstances {
 }
 
 export interface TokenPriceInfo {
-  price: number,
+  price: number;
   name: string;
 }
 
-export interface TokenAllowanceParams {
-  bid?: zAuction.Bid;
-  tokenId?: string;
-  paymentTokenAddress?: string;
+export interface TokenAllowanceByBid {
+  bid: zAuction.Bid;
+  tokenId?: undefined;
+  paymentTokenAddress?: undefined;
+}
+
+interface TokenAllowanceByDomain {
+  tokenId: string;
+  bid?: undefined;
+  paymentTokenAddress?: undefined;
 }
 
+interface TokenAllowanceByPaymentTokenAddress {
+  paymentTokenAddress: string;
+  tokenId?: undefined;
+  bid?: undefined;
+}
+
+interface TokenAllowanceLegacy {
+  tokenId?: undefined;
+  bid?: undefined;
+  paymentTokenAddress?: undefined;
+}
+
+export type TokenAllowanceParams =
```

Comment:
Provide some comments on this type to make sure the user understands how to use it.